scheduler/replica_scheduler.go

```diff
@@ -792,6 +795,32 @@ func (rcs *ReplicaScheduler) IsSchedulableToDisk(size int64, requiredStorage int
 		(size+info.StorageScheduled) <= int64(float64(info.StorageMaximum-info.StorageReserved)*float64(info.OverProvisioningPercentage)/100)
 }
 
+func (rcs *ReplicaScheduler) IsSchedulableToDiskConsiderDiskPressure(diskPressurePercentage, size, requiredStorage int64, info *DiskSchedulingInfo) bool {
+	newDiskUsagePercentage := (requiredStorage + info.StorageScheduled + info.StorageReserved) * 100 / info.StorageMaximum
+	logrus.WithFields(logrus.Fields{
+		"diskUUID":               info.DiskUUID,
+		"diskPressurePercentage": diskPressurePercentage,
+		"requiredStorage":        requiredStorage,
+		"storageScheduled":       info.StorageScheduled,
+		"storageReserved":        info.StorageReserved,
+		"storageMaximum":         info.StorageMaximum,
+	}).Debugf("Evaluated new disk usage percentage after scheduling replica: %v%%", newDiskUsagePercentage)
+
+	return rcs.IsSchedulableToDisk(size, requiredStorage, info) &&
+		newDiskUsagePercentage < int64(diskPressurePercentage)
```

Comment:
Like the split that makes the logic clear.
We can have a code convention rules and checks in the future.
https://go.dev/wiki/CodeReviewComments

cc @innobead